processor_func.go

```diff
@@ -0,0 +1,68 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sdk
+
+import (
+	"context"
+	"errors"
+
+	"github.com/conduitio/conduit-commons/opencdc"
+)
+
+// ProcessorFunc is an adapter allowing use of a function as a Processor.
+type ProcessorFunc struct {
+	UnimplementedProcessor
+
+	specs Specification
+	f     func(context.Context, opencdc.Record) (opencdc.Record, error)
+}
+
+var _ Processor = ProcessorFunc{} // Ensure ProcessorFunc implements Processor
+
+// NewProcessorFunc creates a ProcessorFunc from a function and specifications.
+// This is useful for creating simple processors without needing to implement
+// the full Processor interface.
+func NewProcessorFunc(specs Specification, f func(context.Context, opencdc.Record) (opencdc.Record, error)) ProcessorFunc {
```

Comment:
Instead of specs, maybe just give it a name? Given that there's no configuration, params will be empty. And the version can always be the latest or, if needed, it can be given.

Reply:
I see your point about the configuration parameters, that's valid, but it doesn't feel like that much of a simplification tbh. Only specifying the name isn't _so_ verbose that it would be annoying (`sdk.Specification{Name: "foo"}`), on the other hand we would lose the ability to specify all other fields in the specs. So I'd rather keep the whole specification, keep it flexible 🤷‍♂️